File: prob/451.py
class Solution:
    def frequencySort(self, s: str) -> str:
        freq = {}
        for alpha in s:
            if alpha in freq:
                freq[alpha] += 1
            else:
                freq[alpha] = 1
        li = sorted(freq, key=freq.get, reverse=True)
        # li.sort(key=lambda x: x[1], reverse=True)
        # print(li)
        ans = ""
        for alpha in li:
            ans += alpha * freq[alpha]
        # for [alpha, fr] in li:
        #     ans += alpha*fr
        return ans


def main():
    s1 = Solution()
    word = "tree"
    ans = s1.frequencySort(word)
    print(ans)
    return


if __name__ == '__main__':
    main()

File: prob/451.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
#include<algorithm>
using namespace std;

class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> freq;
        string ans = "";
        for(auto x: s) freq[x]++;
        vector<pair<char, int>> freq_vec(freq.begin(), freq.end());
        // need to check this
        // sort(freq_vec.begin(), freq_vec.end(), )
        for(auto x: freq_vec){
            cout << x.first << ": " << x.second <<endl;
        }
        return ans;
    }
};

int main(){
    Solution s1;
    string s = "tttree";
    string ans = "";
    ans = s1.frequencySort(s);
    cout << ans << endl;
    return 0;
}